jax/numpy/lax_numpy.py

```diff
@@ -158,9 +167,19 @@ def _promote_shapes(*args):
     return args
   else:
     shapes = [shape(arg) for arg in args]
```

Comment:
shapes = map(shape, args)

Reply:
I made it consistent with the line below, but actually I didn't use map and instead used list comprehensions for both because map is 30ns slower on my machine. Gotta pay for those string comparisons somehow!